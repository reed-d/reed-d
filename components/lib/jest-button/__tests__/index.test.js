import rdButton from "../index";
import { mount } from "@vue/test-utils";
import mountTest from "../../../../tests/shared/mountTest";

describe("Button", () => {
  mountTest(rdButton);
  it("测试渲染是否正确", () => {
    const wrapper = mount({
      render() {
        return <rdButton>Follow</rdButton>;
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("create primary button", () => {
    const wrapper = mount({
      render() {
        return <rdButton type="primary">按钮</rdButton>;
      },
    });
    expect(wrapper.find(".type-primary").exists()).toBe(true); // 类名选择器
  });
});
