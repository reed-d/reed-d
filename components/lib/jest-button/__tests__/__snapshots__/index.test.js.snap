// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button class="btn-base type-default">
  <!--v-if-->Follow
</button>
`;

exports[`Button 测试渲染是否正确 1`] = `
<button class="btn-base type-default">
  <!--v-if-->Follow
</button>
`;
